### ISSUE TITLE
Unexpected type with link on keyboard move

### ISSUE THREAD

**Issue by jodator** [ckeditor/ckeditor5#7459]
## 📝 Provide detailed reproduction steps (if any)

1.  Insert `<softBreak>` after a link.
2.  Move selection somewhere else.
3.  Move selection back after the `<softBreak>`.

### ✔️ Expected result

You should type without link (the `linkHref` attribute is not set on selection).

### ❌ Actual result

You can type with link.

![](https://user-images.githubusercontent.com/247363/85034059-f787ef80-b181-11ea-81ee-07e2815d54e3.gif)

## 📃 Other details

*   CKEditor version: latest master

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

**Comment by Reinmar** [comment]
Support for soft breaks was added after the support for two-step caret movement and quite recently in general, so I'd guess we didn't test the editor for this.

**Comment by Reinmar** [comment]
Sorry, I got it now – it only happens when there's no text after the caret. This is a bit special case. Perhaps it's a symmetrical behavior to this:

![](https://user-images.githubusercontent.com/156149/85105487-dfa87e00-b20a-11ea-9e6f-ca5f56bc1259.png)

Note that if there was a text before the caret then you wouldn't have bold enabled:

![](https://user-images.githubusercontent.com/156149/85105530-edf69a00-b20a-11ea-90c9-8145812cb0bc.png)

So, in some cases, the gravity automatically changes to right. Perhaps it somehow leaks to the case with a soft break. As I mentioned, soft breaks were added recently and for a long time we didn't have support for inline elements so I guess we did consider such a case where an inline element needs to stop the search for a neighbour text node.
